components/profile.js

```diff
@@ -65,57 +67,413 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
 
 			switch(i) {
 				case 'name':
-					values.personaName = settings[i];
+					out[8] = {
+						"name": formd[8].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'realName':
-					values.real_name = settings[i];
+					out[9] = {
+						"name": formd[9].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'summary':
-					values.summary = settings[i];
+					out[14] = {
+						"name": formd[14].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'country':
-					values.country = settings[i];
+					out[10] = {
+						"name": formd[10].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'state':
-					values.state = settings[i];
+					out[11] = {
+						"name": formd[11].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'city':
-					values.city = settings[i];
+					out[12] = {
+						"name": formd[12].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'customURL':
-					values.customURL = settings[i];
+					out[13] = {
+						"name": formd[13].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'background':
 					// The assetid of our desired profile background
-					values.profile_background = settings[i];
+					out[15] = {
+						"name": formd[15].name,
+						"value": settings[i]
+					};
 					break;
 
 				case 'featuredBadge':
 					// Currently, game badges aren't supported
-					values.favorite_badge_badgeid = settings[i];
+						out[16] = {
+							"name": formd[16].name,
+							"value": settings[i]
+						};
 					break;
 
 				case 'primaryGroup':
-					if(typeof settings[i] === 'object' && settings[i].getSteamID64) {
-						values.primary_group_steamid = settings[i].getSteamID64();
+					if (typeof settings[i] === 'object' && settings[i].getSteamID64) {
+						out[18] = {
+							"name": formd[18].name,
+							"value": settings[i].getSteamID64()
+						};
 					} else {
-						values.primary_group_steamid = new SteamID(settings[i]).getSteamID64();
+						out[18] = {
+							"name": formd[18].name,
+							"value": new SteamID(settings[i]).getSteamID64()
+						};
 					}
 
 					break;
 
-				// TODO: profile showcases
+				case 'showcases':
+					for (var t in settings[i]) {
+						var showcaseconfig = [false,0,0,[]];
+
+						switch (settings[i][t].showcase) {
+
+							case 'infobox':
+								out.push({
+									"name": "profile_showcase[]",
+									"value": "8"
```

Comment:
Magic number (a lot of them below)

Reply:
What do you mean `magic number`? Are you referring to the value 8? To get those values I tested all show cases on my profile and watched the submitted parameters.

Reply:
A "magic number" in programming is a number with an unclear or unspecified meaning, which leads to very unreadable code and code that's difficult to understand. Usually, a good way to avoid magic numbers is by using enums.